libraries/chain/include/eosio/chain/apply_context.hpp

```diff
@@ -774,10 +774,9 @@ using apply_handler = std::function<void(apply_context&)>;
 
       if( itr == idx.end() ||
           itr->t_id != t_id.id ||
-          !impl::record_scope_compare<IndexType, Scope>::compare(*itr, keys)) return -1;
+          !impl::record_scope_compare<IndexType, Scope>::compare(*itr, keys)) return 0;
```

Comment:
shouldn't we keep the -1 return value? 
How we can distinguish between reading a zero sized record (besides keys) and a "record not found" error.